server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java

```diff
@@ -286,30 +288,30 @@ private void updateBatchFetched(boolean success,
         // has data until then.
         // Method is called inside a transport thread (response listener), so dispatch away
         threadPool.executor(ThreadPool.Names.LOGICAL_REPLICATION).execute(
-            () -> shardReplicationService.getRemoteClusterClient(
-                shardId.getIndex(),
-                client -> RetentionLeaseHelper.renewRetentionLease(
-                    shardId,
-                    toSeqNoReceived,
-                    clusterName,
-                    client,
-                    ActionListener.wrap(
-                        r -> {
-                            // schedule next poll
-                            cancellable = threadPool.schedule(
-                                this::requestBatchToFetch,
-                                replicationSettings.pollDelay(),
-                                ThreadPool.Names.LOGICAL_REPLICATION
-                            );
-                        },
-                        e -> {
-                            LOGGER.warn("Exception renewing retention lease.", e);
-                        }
-                    )
-                ),
-                e -> {
-                }
-            )
+            () -> {
+                shardReplicationService.getRemoteClusterClient(shardId.getIndex())
+                    .whenComplete((client, err) -> {
+                        RetentionLeaseHelper.renewRetentionLease(
```

Comment:
shouldn't this only be done if `err == null`?

Reply:
Right, thanks for catching.